Examples/Swift/ViewController.swift

```diff
@@ -282,37 +282,47 @@ extension ViewController: NavigationMapViewDelegate {
     func navigationMapView(_ mapView: NavigationMapView, didSelect waypoint: Waypoint) {
         guard let routeOptions = currentRoute?.routeOptions else { return }
         let modifiedOptions = routeOptions.without(waypoint: waypoint)
-        
+
         let destroyWaypoint: (UIAlertAction) -> Void = {_ in self.requestRoute(with:modifiedOptions, success: self.defaultSuccess, failure: self.defaultFailure) }
-        
+
         presentWaypointRemovalActionSheet(callback: destroyWaypoint)
     }
-    
+
     func navigationMapView(_ mapView: NavigationMapView, didSelect route: Route) {
         currentRoute = route
     }
-    
+
     private func presentWaypointRemovalActionSheet(callback approve: @escaping ((UIAlertAction) -> Void)) {
         let title = NSLocalizedString("Remove Waypoint?", comment: "Waypoint Removal Action Sheet Title")
         let message = NSLocalizedString("Would you like to remove this waypoint?", comment: "Waypoint Removal Action Sheet Message")
         let removeTitle = NSLocalizedString("Remove Waypoint", comment: "Waypoint Removal Action Item Title")
         let cancelTitle = NSLocalizedString("Cancel", comment: "Waypoint Removal Action Sheet Cancel Item Title")
-        
+
         let actionSheet = UIAlertController(title: title, message: message, preferredStyle: .actionSheet)
         let remove = UIAlertAction(title: removeTitle, style: .destructive, handler: approve)
         let cancel = UIAlertAction(title: cancelTitle, style: .cancel, handler: nil)
         [remove, cancel].forEach(actionSheet.addAction(_:))
-        
+
         self.present(actionSheet, animated: true, completion: nil)
     }
+
+    // Called when there is an error with speaking a voice instruction.
+    func voiceController(_ voiceController: RouteVoiceController, spokenInstructionsDidFailWith error: Error) {
+        print(error.localizedDescription)
+    }
+    
+    // Called when an instruction is interrupted by a new voice instruction.
+    func voiceController(_ voiceController: RouteVoiceController, didInterrupt: String, with instruction: String) {
```

Comment:
This method should accept two Instruction objects as parameters, not just strings. That way the implementation can include logic based on the distance between the instructions, for example.